In [1]:
import os
import json
import pandas as pd

base_path = "/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline"
output_file = "/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/res_gpt_4o_aide.csv"

data_list = []

for folder in os.listdir(base_path):
    folder_path = os.path.join(base_path, folder)
    if os.path.isdir(folder_path):
        grading_file_path = None
        for file in os.listdir(folder_path):
            if file.endswith("grading_report.json"):
                grading_file_path = os.path.join(folder_path, file)
                break

        with open(grading_file_path, "r") as f:
            grading_data = json.load(f)

            for report in grading_data.get("competition_reports", []):
                data_list.append({
                    "competition_id": report.get("competition_id"),
                    "score": report.get("score"),
                    "gold_threshold": report.get("gold_threshold"),
                    "silver_threshold": report.get("silver_threshold"),
                    "bronze_threshold": report.get("bronze_threshold"),
                    "median_threshold": report.get("median_threshold"),
                    "any_medal": report.get("any_medal"),
                    "gold_medal": report.get("gold_medal"),
                    "silver_medal": report.get("silver_medal"),
                    "bronze_medal": report.get("bronze_medal"),
                    "above_median": report.get("above_median"),
                    "submission_exists": report.get("submission_exists"),
                    "valid_submission": report.get("valid_submission"),
                })

competition_df = pd.DataFrame(data_list)
competition_df


competition_id    score  gold_threshold  \
0                        tweet-sentiment-extraction  0.59324         0.72689   
1                            champs-scalar-coupling  2.11599        -2.87509   
2     jigsaw-toxic-comment-classification-challenge  0.98014         0.98740   
3            detecting-insults-in-social-commentary      NaN         0.83321   
4                tabular-playground-series-dec-2021  0.96167         0.95660   
...                                             ...      ...             ...   
2906       cdiscount-image-classification-challenge      NaN         0.77604   
2907       cdiscount-image-classification-challenge  0.05001         0.77604   
2908       cdiscount-image-classification-challenge      NaN         0.77604   
2909       cdiscount-image-classification-challenge      NaN         0.77604   
2910       cdiscount-image-classification-challenge      NaN         0.77604   

      silver_threshold  bronze_threshold  median_threshold  any_medal  \
0              0.71752           0.71705           0.71378      False   
1             -2.03119          -1.90122          -0.95290      False   
2              0.98668           0.98639           0.98079      False   
3              0.82307           0.79111           0.77842      False   
4              0.95658           0.95658           0.95342       True   
...                ...               ...               ...        ...   
2906           0.73526           0.70898           0.44055      False   
2907           0.73526           0.70898           0.44055      False   
2908           0.73526           0.70898           0.44055      False   
2909           0.73526           0.70898           0.44055      False   
2910           0.73526           0.70898           0.44055      False   

      gold_medal  silver_medal  bronze_medal  above_median  submission_exists  \
0          False         False         False         False               True   
1          False         False         False         False               True   
2          False         False         False         False               True   
3          False         False         False         False               True   
4           True         False         False          True               True   
...          ...           ...           ...           ...                ...   
2906       False         False         False         False               True   
2907       False         False         False         False               True   
2908       False         False         False         False               True   
2909       False         False         False         False               True   
2910       False         False         False         False              False   

      valid_submission  
0                 True  
1                 True  
2                 True  
3                False  
4                 True  
...                ...  
2906             False  
2907              True  
2908             False  
2909             False  
2910             False  

[2911 rows x 13 columns]

In [2]:
grouped_df = competition_df.groupby('competition_id').agg(
    total_competitions=('competition_id', 'size'),
    valid_submissions=('valid_submission', 'sum'),
    submission_exists=('submission_exists', 'sum'),
    any_medal=('any_medal', 'sum'),
    gold_medal=('gold_medal', 'sum'),
    silver_medal=('silver_medal', 'sum'),
    bronze_medal=('bronze_medal', 'sum'),
    above_median=('above_median', 'sum')
).reset_index()
# grouped_df.to_csv("/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/res_gpt_4o_aide_each.csv", index=False)
grouped_df

competition_id  total_competitions  \
0     3d-object-detection-for-autonomous-vehicles                  39   
1                                         AI4Code                  45   
2                    aerial-cactus-identification                  39   
3                      alaska2-image-steganalysis                  39   
4                   aptos2019-blindness-detection                  38   
..                                            ...                 ...   
70         uw-madison-gi-tract-image-segmentation                  38   
71                 ventilator-pressure-prediction                  39   
72               vesuvius-challenge-ink-detection                  39   
73  vinbigdata-chest-xray-abnormalities-detection                  37   
74                whale-categorization-playground                  38   

    valid_submissions  submission_exists  any_medal  gold_medal  silver_medal  \
0                  19                 33          0           0             0   
1                  19                 29          0           0             0   
2                  22                 36          5           5             0   
3                   4                 35          0           0             0   
4                  35                 35          0           0             0   
..                ...                ...        ...         ...           ...   
70                  5                 22          0           0             0   
71                 36                 36          0           0             0   
72                 16                 28          0           0             0   
73                  0                  0          0           0             0   
74                 34                 34          1           0             0   

    bronze_medal  above_median  
0              0             0  
1              0             0  
2              0            21  
3              0             0  
4              0            11  
..           ...           ...  
70             0             0  
71             0             0  
72             0             0  
73             0             0  
74             1             2  

[75 rows x 9 columns]

In [3]:
selected_competitions = [
    'aerial-cactus-identification',
    'aptos2019-blindness-detection',
    'denoising-dirty-documents',
    'detecting-insults-in-social-commentary',
    'dog-breed-identification',
    'dogs-vs-cats-redux-kernels-edition',
    'histopathologic-cancer-detection',
    'jigsaw-toxic-comment-classification-challenge',
    'leaf-classification',
    'mlsp-2013-birds',
    'new-york-city-taxi-fare-prediction',
    'nomad2018-predict-transparent-conductors',
    'plant-pathology-2020-fgvc7',
    'random-acts-of-pizza',
    'ranzcr-clip-catheter-line-classification',
    'siim-isic-melanoma-classification',
    'spooky-author-identification',
    'tabular-playground-series-dec-2021',
    'tabular-playground-series-may-2022',
    'text-normalization-challenge-english-language',
    'text-normalization-challenge-russian-language',
    'the-icml-2013-whale-challenge-right-whale-redux',
]

filtered_df = grouped_df[grouped_df['competition_id'].isin(selected_competitions)]
filtered_df

competition_id  total_competitions  \
2                      aerial-cactus-identification                  39   
4                     aptos2019-blindness-detection                  38   
11                        denoising-dirty-documents                  39   
12           detecting-insults-in-social-commentary                  37   
13                         dog-breed-identification                  37   
14               dogs-vs-cats-redux-kernels-edition                  39   
23                 histopathologic-cancer-detection                  39   
32    jigsaw-toxic-comment-classification-challenge                  39   
35                              leaf-classification                  38   
38                                  mlsp-2013-birds                  38   
40               new-york-city-taxi-fare-prediction                  38   
42         nomad2018-predict-transparent-conductors                  39   
45                       plant-pathology-2020-fgvc7                  39   
48                             random-acts-of-pizza                  39   
49         ranzcr-clip-catheter-line-classification                  39   
55                siim-isic-melanoma-classification                  39   
57                     spooky-author-identification                  38   
60               tabular-playground-series-dec-2021                  39   
61               tabular-playground-series-may-2022                  39   
64    text-normalization-challenge-english-language                  38   
65    text-normalization-challenge-russian-language                  39   
67  the-icml-2013-whale-challenge-right-whale-redux                  39   

    valid_submissions  submission_exists  any_medal  gold_medal  silver_medal  \
2                  22                 36          5           5             0   
4                  35                 35          0           0             0   
11                  0                 37          0           0             0   
12                  0                 34          0           0             0   
13                 29                 34          0           0             0   
14                 31                 34         13          11             2   
23                 17                 36         14          13             0   
32                 34                 34          0           0             0   
35                 34                 35          0           0             0   
38                  4                 12          1           0             0   
40                 28                 29          0           0             0   
42                 34                 36         24           0            19   
45                 35                 35         29          18            10   
48                 36                 36          3           0             3   
49                 29                 33          0           0             0   
55                 21                 25          0           0             0   
57                 34                 35          1           0             0   
60                 36                 36         34          34             0   
61                 30                 31          0           0             0   
64                 31                 31         15           0             0   
65                 12                 12          0           0             0   
67                 34                 36         22           0            17   

    bronze_medal  above_median  
2              0            21  
4              0            11  
11             0             0  
12             0             0  
13             0             3  
14             0            20  
23             1            14  
32             0            12  
35             0             2  
38             1             1  
40             0             0  
42             5            25  
45             1  

In [7]:
filtered_df.to_csv("/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_mini_baseline/mlebench_aide_lite_results_from_official_repo.csv", index=False)

In [4]:
# Made Submission Rate
total_competitions = filtered_df['total_competitions'].sum()

comp_made_rate = filtered_df['submission_exists'].sum() / total_competitions
print("comp_made_rate:", comp_made_rate)

# Valid Submission Rate
comp_finished_rate = filtered_df['valid_submissions'].sum() / total_competitions
print("comp_finished_rate:", comp_finished_rate)

# Above Median Rate
above_median_rate = filtered_df['above_median'].sum() / total_competitions
print("above_median_rate:", above_median_rate)

# Gold Medal Rate
gold_medal_rate = filtered_df['gold_medal'].sum() / total_competitions
print("gold_medal_rate:", gold_medal_rate)

# Silver Medal Rate
silver_medal_rate = filtered_df['silver_medal'].sum() / total_competitions
print("silver_medal_rate:", silver_medal_rate)

# Bronze Medal Rate
bronze_medal_rate = filtered_df['bronze_medal'].sum() / total_competitions
print("bronze_medal_rate:", bronze_medal_rate)

# Any Medal Rate
any_medal_rate = filtered_df['any_medal'].sum() / total_competitions
print("any_medal_rate:", any_medal_rate)

markdown_table = f"""
| AIDE (OpenAI, gpt-4o_2024-08-06) | {comp_made_rate*100:.1f}             | {comp_finished_rate*100:.1f}               | {above_median_rate*100:.1f}           | {bronze_medal_rate*100:.1f}      | {silver_medal_rate*100:.1f}      | {gold_medal_rate*100:.1f}    | {any_medal_rate*100:.1f}         |
"""

print(markdown_table)

comp_made_rate: 0.8278301886792453
comp_finished_rate: 0.6674528301886793
above_median_rate: 0.30424528301886794
gold_medal_rate: 0.09551886792452831
silver_medal_rate: 0.060141509433962265
bronze_medal_rate: 0.03419811320754717
any_medal_rate: 0.18985849056603774

| AIDE (OpenAI, gpt-4o_2024-08-06) | 82.8             | 66.7               | 30.4           | 3.4      | 6.0      | 9.6    | 19.0         |



In [6]:
grouped_df = filtered_df

grouped_df = competition_df.groupby('competition_id').agg(
    total_competitions=('competition_id', 'size'),
    valid_submissions=('valid_submission', 'sum'),
    submission_exists=('submission_exists', 'sum'),
    any_medal=('any_medal', 'sum'),
    gold_medal=('gold_medal', 'sum'),
    silver_medal=('silver_medal', 'sum'),
    bronze_medal=('bronze_medal', 'sum'),
    above_median=('above_median', 'sum')
).reset_index()

# Convert counts to percentages
grouped_df['valid_submissions'] = (grouped_df['valid_submissions'] / grouped_df['total_competitions']) * 100
grouped_df['submission_exists'] = (grouped_df['submission_exists'] / grouped_df['total_competitions']) * 100
grouped_df['any_medal'] = (grouped_df['any_medal'] / grouped_df['total_competitions']) * 100
grouped_df['gold_medal'] = (grouped_df['gold_medal'] / grouped_df['total_competitions']) * 100
grouped_df['silver_medal'] = (grouped_df['silver_medal'] / grouped_df['total_competitions']) * 100
grouped_df['bronze_medal'] = (grouped_df['bronze_medal'] / grouped_df['total_competitions']) * 100
grouped_df['above_median'] = (grouped_df['above_median'] / grouped_df['total_competitions']) * 100
grouped_df.to_csv("/data/userdata/v-yuanteli/mlebench-office/mle-bench/runs_aide_mini_baseline/res_gpt_4o_aide_each2.csv", index=False)
grouped_df

competition_id  total_competitions  \
0     3d-object-detection-for-autonomous-vehicles                  39   
1                                         AI4Code                  45   
2                    aerial-cactus-identification                  39   
3                      alaska2-image-steganalysis                  39   
4                   aptos2019-blindness-detection                  38   
..                                            ...                 ...   
70         uw-madison-gi-tract-image-segmentation                  38   
71                 ventilator-pressure-prediction                  39   
72               vesuvius-challenge-ink-detection                  39   
73  vinbigdata-chest-xray-abnormalities-detection                  37   
74                whale-categorization-playground                  38   

    valid_submissions  submission_exists  any_medal  gold_medal  silver_medal  \
0           48.717949          84.615385   0.000000    0.000000           0.0   
1           42.222222          64.444444   0.000000    0.000000           0.0   
2           56.410256          92.307692  12.820513   12.820513           0.0   
3           10.256410          89.743590   0.000000    0.000000           0.0   
4           92.105263          92.105263   0.000000    0.000000           0.0   
..                ...                ...        ...         ...           ...   
70          13.157895          57.894737   0.000000    0.000000           0.0   
71          92.307692          92.307692   0.000000    0.000000           0.0   
72          41.025641          71.794872   0.000000    0.000000           0.0   
73           0.000000           0.000000   0.000000    0.000000           0.0   
74          89.473684          89.473684   2.631579    0.000000           0.0   

    bronze_medal  above_median  
0       0.000000      0.000000  
1       0.000000      0.000000  
2       0.000000     53.846154  
3       0.000000      0.000000  
4       0.000000     28.947368  
..           ...           ...  
70      0.000000      0.000000  
71      0.000000      0.000000  
72      0.000000      0.000000  
73      0.000000      0.000000  
74      2.631579      5.263158  

[75 rows x 9 columns]

In [5]:
# Made Submission Rate
total_competitions = filtered_df['total_competitions'].sum()

comp_made_rate = filtered_df['submission_exists'].sum() / total_competitions
print("comp_made_rate:", comp_made_rate)

# Valid Submission Rate
comp_finished_rate = filtered_df['valid_submissions'].sum() / total_competitions
print("comp_finished_rate:", comp_finished_rate)

# Above Median Rate
above_median_rate = filtered_df['above_median'].sum() / total_competitions
print("above_median_rate:", above_median_rate)

# Gold Medal Rate
gold_medal_rate = filtered_df['gold_medal'].sum() / total_competitions
print("gold_medal_rate:", gold_medal_rate)

# Silver Medal Rate
silver_medal_rate = filtered_df['silver_medal'].sum() / total_competitions
print("silver_medal_rate:", silver_medal_rate)

# Bronze Medal Rate
bronze_medal_rate = filtered_df['bronze_medal'].sum() / total_competitions
print("bronze_medal_rate:", bronze_medal_rate)

# Any Medal Rate
any_medal_rate = filtered_df['any_medal'].sum() / total_competitions
print("any_medal_rate:", any_medal_rate)

markdown_table = f"""
| AIDE (OpenAI, gpt-4o_2024-08-06) | {comp_made_rate*100:.1f}             | {comp_finished_rate*100:.1f}               | {above_median_rate*100:.1f}           | {bronze_medal_rate*100:.1f}      | {silver_medal_rate*100:.1f}      | {gold_medal_rate*100:.1f}    | {any_medal_rate*100:.1f}         |
"""

print(markdown_table)

comp_made_rate: 0.8278301886792453
comp_finished_rate: 0.6674528301886793
above_median_rate: 0.30424528301886794
gold_medal_rate: 0.09551886792452831
silver_medal_rate: 0.060141509433962265
bronze_medal_rate: 0.03419811320754717
any_medal_rate: 0.18985849056603774

| AIDE (OpenAI, gpt-4o_2024-08-06) | 82.8             | 66.7               | 30.4           | 3.4      | 6.0      | 9.6    | 19.0         |



In [2]:
import pandas as pd

path1="/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/res_gpt_4o_aide_each2.csv"
path2="/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/our_result_gpt_4o_aide_each2.csv"
df1 = pd.read_csv(path1)
df2 = pd.read_csv(path2)

df1 = df1.add_prefix('OpenAI_')
df2 = df2.add_prefix('Ours_')

df1 = df1.rename(columns={'OpenAI_competition_id': 'competition_id'})
df2 = df2.rename(columns={'Ours_competition_id': 'competition_id'})

merged_df = pd.merge(df1, df2, on='competition_id')
merged_df = merged_df[[merged_df.columns[0]] + [col for pair in zip(df1.columns[1:], df2.columns[1:]) for col in pair]]
merged_df.to_csv("/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/compare2.csv", index=False)

In [9]:
import seaborn as sns
import matplotlib.pyplot as plt
import pandas as pd

competition_size_df = pd.read_csv("/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/competition_size.csv")
merged_df_with_size = pd.merge(merged_df, competition_size_df, on='competition_id')
merged_df_with_size.sort_values('size_MB', ascending=False, inplace=True)
heatmap_data_without_size = merged_df_with_size[['competition_id', 'OpenAI_valid_submissions', 'Ours_valid_submissions', 'OpenAI_submission_exists', 'Ours_submission_exists', 'size_MB']]
# heatmap_data_without_size.shape
heatmap_data_without_size = heatmap_data_without_size.iloc[-15:].set_index('competition_id')
heatmap_data_without_size = heatmap_data_without_size.sort_index()
heatmap_data_without_size.index
# heatmap_data_without_size = merged_df_with_size[['competition_id', 'OpenAI_valid_submissions', 'Ours_valid_submissions', 'OpenAI_submission_exists', 'Ours_submission_exists']]

# heatmap_data_without_size.set_index('competition_id', inplace=True)

# fig, ax = plt.subplots(1, 2, figsize=(20, 24), gridspec_kw={'width_ratios': [4, 1]})

# sns.heatmap(heatmap_data_without_size, annot=True, cmap='coolwarm', fmt=".2f", ax=ax[0])
# ax[0].set_title('Heatmap of Valid Submissions and Submission Exists')

# sns.barplot(y='competition_id', x='size_MB', data=merged_df_with_size, ax=ax[1])
# ax[1].set_title('Competition Dataset Size (MB)')
# ax[1].set_xlabel('Size (MB)')
# ax[1].set_yticklabels([])
# ax[1].spines['top'].set_visible(False)
# ax[1].spines['right'].set_visible(False)
# ax[1].spines['left'].set_visible(False)
# ax[1].spines['bottom'].set_visible(False)

# plt.tight_layout()
# plt.show()


Index(['aerial-cactus-identification',
       'chaii-hindi-and-tamil-question-answering',
       'detecting-insults-in-social-commentary', 'google-quest-challenge',
       'jigsaw-toxic-comment-classification-challenge', 'leaf-classification',
       'learning-agency-lab-automated-essay-scoring-2',
       'nomad2018-predict-transparent-conductors', 'random-acts-of-pizza',
       'spooky-author-identification', 'stanford-covid-vaccine',
       'statoil-iceberg-classifier-challenge',
       'tgs-salt-identification-challenge', 'tweet-sentiment-extraction',
       'us-patent-phrase-to-phrase-matching'],
      dtype='object', name='competition_id')

In [ ]:
grouped_df = competition_df.groupby('competition_id').agg(
    total_competitions=('competition_id', 'size'),
    valid_submissions=('valid_submission', 'sum'),
    submission_exists=('submission_exists', 'sum'),
    any_medal=('any_medal', 'sum'),
    gold_medal=('gold_medal', 'sum'),
    silver_medal=('silver_medal', 'sum'),
    bronze_medal=('bronze_medal', 'sum'),
    above_median=('above_median', 'sum')
).reset_index()

# Convert counts to percentages
grouped_df['valid_submissions'] = (grouped_df['valid_submissions'] / grouped_df['total_competitions']) * 100
grouped_df['submission_exists'] = (grouped_df['submission_exists'] / grouped_df['total_competitions']) * 100
grouped_df['any_medal'] = (grouped_df['any_medal'] / grouped_df['total_competitions']) * 100
grouped_df['gold_medal'] = (grouped_df['gold_medal'] / grouped_df['total_competitions']) * 100
grouped_df['silver_medal'] = (grouped_df['silver_medal'] / grouped_df['total_competitions']) * 100
grouped_df['bronze_medal'] = (grouped_df['bronze_medal'] / grouped_df['total_competitions']) * 100
grouped_df['above_median'] = (grouped_df['above_median'] / grouped_df['total_competitions']) * 100
grouped_df.to_csv("/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/res_gpt_4o_aide_each2.csv", index=False)
grouped_df

competition_id  total_competitions  \
0     3d-object-detection-for-autonomous-vehicles                  39   
1                                         AI4Code                  45   
2                    aerial-cactus-identification                  39   
3                      alaska2-image-steganalysis                  39   
4                   aptos2019-blindness-detection                  38   
..                                            ...                 ...   
70         uw-madison-gi-tract-image-segmentation                  38   
71                 ventilator-pressure-prediction                  39   
72               vesuvius-challenge-ink-detection                  39   
73  vinbigdata-chest-xray-abnormalities-detection                  37   
74                whale-categorization-playground                  38   

    valid_submissions  submission_exists  any_medal  gold_medal  silver_medal  \
0           48.717949          84.615385   0.000000    0.000000           0.0   
1           42.222222          64.444444   0.000000    0.000000           0.0   
2           56.410256          92.307692  12.820513   12.820513           0.0   
3           10.256410          89.743590   0.000000    0.000000           0.0   
4           92.105263          92.105263   0.000000    0.000000           0.0   
..                ...                ...        ...         ...           ...   
70          13.157895          57.894737   0.000000    0.000000           0.0   
71          92.307692          92.307692   0.000000    0.000000           0.0   
72          41.025641          71.794872   0.000000    0.000000           0.0   
73           0.000000           0.000000   0.000000    0.000000           0.0   
74          89.473684          89.473684   2.631579    0.000000           0.0   

    bronze_medal  above_median  
0       0.000000      0.000000  
1       0.000000      0.000000  
2       0.000000     53.846154  
3       0.000000      0.000000  
4       0.000000     28.947368  
..           ...           ...  
70      0.000000      0.000000  
71      0.000000      0.000000  
72      0.000000      0.000000  
73      0.000000      0.000000  
74      2.631579      5.263158  

[75 rows x 9 columns]

In [2]:
competition_df.to_csv(output_file, index=False)

In [3]:
# Made Submission Rate
comp_made_rate = competition_df['submission_exists'].sum() / len(competition_df)
print("comp_made_rate:", comp_made_rate)

# Valid Submission Rate
comp_finished_rate = competition_df['valid_submission'].sum() / len(competition_df)
print("comp_finished_rate:", comp_finished_rate)

# Above Median Rate
above_median_rate = competition_df['above_median'].sum() / len(competition_df)
print("above_median_rate:", above_median_rate)

# Gold Medal Rate
gold_medal_rate = competition_df['gold_medal'].sum() / len(competition_df)
print("gold_medal_rate:", gold_medal_rate)

# Silver Medal Rate
silver_medal_rate = competition_df['silver_medal'].sum() / len(competition_df)
print("silver_medal_rate:", silver_medal_rate)

# Bronze Medal Rate
bronze_medal_rate = competition_df['bronze_medal'].sum() / len(competition_df)
print("bronze_medal_rate:", bronze_medal_rate)

# Any Medal Rate
any_medal_rate = competition_df['any_medal'].sum() / len(competition_df)
print("any_medal_rate:", any_medal_rate)



comp_made_rate: 0.7128134661628306
comp_finished_rate: 0.556166265888011
above_median_rate: 0.1456544142906218
gold_medal_rate: 0.051185159738921335
silver_medal_rate: 0.022329096530401923
bronze_medal_rate: 0.01511508072827207
any_medal_rate: 0.08862933699759533


## mle-bench full Dataset
| Model                          | Made Submission (%) | Valid Submission (%) | Above Median (%) | Bronze (%) | Silver (%) | Gold (%) | Any Medal (%) |
|--------------------------------|---------------------|-----------------------|------------------|------------|------------|----------|---------------|
| OpenAI (gpt-4o-2024-08-06, Paper)     | 70.7 ± 0.9          | 54.9 ± 1.0            | 14.4 ± 0.7       | 1.6 ± 0.2  | 2.2 ± 0.3  | 5.0 ± 0.4 | 8.7 ± 0.5    |
| OpenAI (gpt-4o-2024-08-06, GitHub Repo)     | 71.2          | 55.6            | 14.6       | 1.5  | 2.2  | 5.1 | 8.9    |
| OpenAI (gpt-4o-2024-08-06, GitHub Repo with SE)     | 71.3 ± 0.8  | 55.6 ± 0.9  | 14.6 ± 0.7  | 1.5 ± 0.2  | 2.2 ± 0.3  | 5.1 ± 0.4  | 8.9 ± 0.5  |


| Ours (gpt-4o)                  | 53.3                | 48.0                  | 13.3             | 0.0        | 2.7        | 2.7      | 5.3           |
| Ours (gpt-4o-ad)               | 49.3                | 40.0                  | 9.3              | 1.3        | 2.6        | 4.0      | 8.0           |


## mle-bench lite dataset
| Model                          | Made Submission (%) | Valid Submission (%) | Above Median (%) | Bronze (%) | Silver (%) | Gold (%) | Any Medal (%) |
|--------------------------------|---------------------|-----------------------|------------------|------------|------------|----------|---------------|
| AIDE (OpenAI, gpt-4o_2024-08-06, 24h) | 82.8             | 66.7               | 30.4           | 3.4      | 6.0      | 9.6    | 19.0         |
| AIDE Reproduce (gpt-4o, 24h)                  | 81.8                | 63.6                  | 18.2             | 0.0        | 9.1        | 0.0      | 9.1           |
| AIDE Reproduce (o1, 24h)                  | 100.0               | 100.0                  | 54.5             | 9.1        | 9.1        | 18.2      | 36.4           |
| AIDE Reproduce (gpt-4o, 11h)                  | 77.3                | 63.6                  | 18.2             | 0.0        | 4.5        | 4.5      | 9.1           |
| RD-Agent (gpt-4o, 11h)                  | 90.9                | 90.9                  | 31.8             | 9.1        | 0.0        | 13.6      | 22.7           |

{'any_medal': 9.1,
 'gold_medal': 4.5,
 'silver_medal': 4.5,
 'bronze_medal': 0.0,
 'above_median': 18.2,
 'submission_exists': 77.3,
 'valid_submission': 63.6}

In [ ]:
import pandas as pd
metrics = ["any_medal", "gold_medal", "silver_medal", "bronze_medal", "above_median", "submission_exists", "valid_submission"]
df = pd.read_csv('aide_gpt_4o_our_results.csv')
df['relative_time'] = pd.to_datetime(df['times']) - pd.to_datetime(df['times'].min())

def compute_ratio_within_time(df, metric, hours):
    df_time = df[df['relative_time'] <= pd.Timedelta(hours=hours)]
    if df_time.empty:
        return 0

    last_entries = df_time.groupby('competition').apply(lambda x: x.loc[x['relative_time'].idxmax()])
    return last_entries[metric].sum() / last_entries.shape[0] if last_entries.shape[0] > 0 else 0  # 避免除零错误

aide_ratios_11h = {metric: round(compute_ratio_within_time(df, metric, 11) * 100, 1) for metric in metrics}
aide_ratios_11h

/tmp/ipykernel_2696264/1093152214.py:4: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  df['relative_time'] = pd.to_datetime(df['times']) - pd.to_datetime(df['times'].min())
/tmp/ipykernel_2696264/1093152214.py:11: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will be excluded from the operation. Either pass `include_groups=False` to exclude the groupings or explicitly select the grouping columns after groupby to silence this warning.
  last_entries = df_time.groupby('competition').apply(lambda x: x.loc[x['relative_time'].idxmax()])
/tmp/ipykernel_2696264/1093152214.py:11: DeprecationWarning: DataFrameGroupBy.apply operated on the grouping columns. This behavior is deprecated, and in a future version of pandas the grouping columns will b

{'any_medal': 9.5,
 'gold_medal': 4.8,
 'silver_medal': 4.8,
 'bronze_medal': 0.0,
 'above_median': 19.0,
 'submission_exists': 81.0,
 'valid_submission': 66.7}

In [8]:
output_file = "/home/v-yuanteli/mlebench-office/mle-bench/runs_aide_baseline/res_gpt_4o_aide_each.csv"
output_df = pd.read_csv(output_file)
output_df

competition_id  total_competitions  \
0     3d-object-detection-for-autonomous-vehicles                  39   
1                                         AI4Code                  45   
2                    aerial-cactus-identification                  39   
3                      alaska2-image-steganalysis                  39   
4                   aptos2019-blindness-detection                  38   
..                                            ...                 ...   
70         uw-madison-gi-tract-image-segmentation                  38   
71                 ventilator-pressure-prediction                  39   
72               vesuvius-challenge-ink-detection                  39   
73  vinbigdata-chest-xray-abnormalities-detection                  37   
74                whale-categorization-playground                  38   

    valid_submissions  submission_exists  any_medal  gold_medal  silver_medal  \
0                  19                 33          0           0             0   
1                  19                 29          0           0             0   
2                  22                 36          5           5             0   
3                   4                 35          0           0             0   
4                  35                 35          0           0             0   
..                ...                ...        ...         ...           ...   
70                  5                 22          0           0             0   
71                 36                 36          0           0             0   
72                 16                 28          0           0             0   
73                  0                  0          0           0             0   
74                 34                 34          1           0             0   

    bronze_medal  above_median  
0              0             0  
1              0             0  
2              0            21  
3              0             0  
4              0            11  
..           ...           ...  
70             0             0  
71             0             0  
72             0             0  
73             0             0  
74             1             2  

[75 rows x 9 columns]

In [10]:
filtered_df  = output_df

total_competitions = filtered_df['total_competitions'].sum()

comp_made_rate = filtered_df['submission_exists'].sum() / total_competitions
print("comp_made_rate:", comp_made_rate)

# Valid Submission Rate
comp_finished_rate = filtered_df['valid_submissions'].sum() / total_competitions
print("comp_finished_rate:", comp_finished_rate)

# Above Median Rate
above_median_rate = filtered_df['above_median'].sum() / total_competitions
print("above_median_rate:", above_median_rate)

# Gold Medal Rate
gold_medal_rate = filtered_df['gold_medal'].sum() / total_competitions
print("gold_medal_rate:", gold_medal_rate)

# Silver Medal Rate
silver_medal_rate = filtered_df['silver_medal'].sum() / total_competitions
print("silver_medal_rate:", silver_medal_rate)

# Bronze Medal Rate
bronze_medal_rate = filtered_df['bronze_medal'].sum() / total_competitions
print("bronze_medal_rate:", bronze_medal_rate)

# Any Medal Rate
any_medal_rate = filtered_df['any_medal'].sum() / total_competitions
print("any_medal_rate:", any_medal_rate)


comp_made_rate: 0.7128134661628306
comp_finished_rate: 0.556166265888011
above_median_rate: 0.1456544142906218
gold_medal_rate: 0.051185159738921335
silver_medal_rate: 0.022329096530401923
bronze_medal_rate: 0.01511508072827207
any_medal_rate: 0.08862933699759533


In [ ]:
import numpy as np

# 计算总比赛次数
total_competitions = filtered_df['total_competitions'].sum()

# 计算各项比率
metrics = {
    "comp_made_rate": filtered_df['submission_exists'].sum() / total_competitions,
    "comp_finished_rate": filtered_df['valid_submissions'].sum() / total_competitions,
    "above_median_rate": filtered_df['above_median'].sum() / total_competitions,
    "gold_medal_rate": filtered_df['gold_medal'].sum() / total_competitions,
    "silver_medal_rate": filtered_df['silver_medal'].sum() / total_competitions,
    "bronze_medal_rate": filtered_df['bronze_medal'].sum() / total_competitions,
    "any_medal_rate": filtered_df['any_medal'].sum() / total_competitions,
}

# 计算标准误差（SE）: SE = sqrt(p(1-p)/n)
se_metrics = {
    key: np.sqrt(value * (1 - value) / total_competitions) for key, value in metrics.items()
}

# 输出计算结果，格式为 “均值 ± 标准误差”
for key, value in metrics.items():
    se = se_metrics[key]
    print(f"{key}: {value:.6f} ± {se:.6f}")

markdown_table = f"""
| AIDE (OpenAI, gpt-4o_2024-08-06) | {comp_made_rate*100:.1f}             | {comp_finished_rate*100:.1f}               | {above_median_rate*100:.1f}           | {bronze_medal_rate*100:.1f}      | {silver_medal_rate*100:.1f}      | {gold_medal_rate*100:.1f}    | {any_medal_rate*100:.1f}         |
"""



comp_made_rate: 0.712813 ± 0.008386
comp_finished_rate: 0.556166 ± 0.009209
above_median_rate: 0.145654 ± 0.006538
gold_medal_rate: 0.051185 ± 0.004085
silver_medal_rate: 0.022329 ± 0.002738
bronze_medal_rate: 0.015115 ± 0.002261
any_medal_rate: 0.088629 ± 0.005268


In [16]:
markdown_table = f"""
| OpenAI (gpt-4o-2024-08-06, GitHub Repo) | Competition Made (%) | Competition Finished (%) | Above Median (%) | Bronze Medal (%) | Silver Medal (%) | Gold Medal (%) | Any Medal (%) |
|----------------------------------|--------------------|----------------------|----------------|----------------|----------------|--------------|--------------|
| AIDE                            | {metrics['comp_made_rate']*100:.1f} ± {se_metrics['comp_made_rate']*100:.1f}  | {metrics['comp_finished_rate']*100:.1f} ± {se_metrics['comp_finished_rate']*100:.1f}  | {metrics['above_median_rate']*100:.1f} ± {se_metrics['above_median_rate']*100:.1f}  | {metrics['bronze_medal_rate']*100:.1f} ± {se_metrics['bronze_medal_rate']*100:.1f}  | {metrics['silver_medal_rate']*100:.1f} ± {se_metrics['silver_medal_rate']*100:.1f}  | {metrics['gold_medal_rate']*100:.1f} ± {se_metrics['gold_medal_rate']*100:.1f}  | {metrics['any_medal_rate']*100:.1f} ± {se_metrics['any_medal_rate']*100:.1f}  |
"""

print(markdown_table)


| OpenAI (gpt-4o-2024-08-06, GitHub Repo) | Competition Made (%) | Competition Finished (%) | Above Median (%) | Bronze Medal (%) | Silver Medal (%) | Gold Medal (%) | Any Medal (%) |
|----------------------------------|--------------------|----------------------|----------------|----------------|----------------|--------------|--------------|
| AIDE                            | 71.3 ± 0.8  | 55.6 ± 0.9  | 14.6 ± 0.7  | 1.5 ± 0.2  | 2.2 ± 0.3  | 5.1 ± 0.4  | 8.9 ± 0.5  |



---